src/main/java/com/odysseusinc/arachne/datanode/service/messaging/CohortCharacterizationRequestHandler.java

```diff
@@ -1,166 +1,105 @@
-                        }
-                        if (summaryEnabled) {
-                            files.add(generateFile("cohort/cohort-summary.sql", "cohort-summary.ohdsi.sql"));
-                        }
-                    } catch (IOException e) {
-                        logger.error("Failed to build CC data", e);
-                        throw new RuntimeIOException("Failed to build CC data", e);
-                    }
-
-                } else {
-                    return null;
-                }
-            }
-            return files;
-        }).orElse(null);
-    }
-
-    private MockMultipartFile generateFile(String resourcePath, String outputName) throws IOException {
-
-        ClassPathResource resource = new ClassPathResource(resourcePath);
-        final String resourceContent = IOUtils.toString(resource.getInputStream(), Charset.defaultCharset());
-        return new MockMultipartFile(outputName, ignorePreprocessingMark(resourceContent).getBytes());
-    }
-
-    private MultipartFile getRunner(String initialFileName) throws IOException {
-
-        Map<String, Object> params = new HashMap<>();
-        params.put("initialFileName", initialFileName);
-        String result = runnerTemplate.apply(params);
-        return new MockMultipartFile("main.r", result.getBytes());
-    }
-
-    @Override
-    public CommonAnalysisType getAnalysisType() {
-
-        return CommonAnalysisType.COHORT_CHARACTERIZATION;
-    }
-
-    @Override
-    public void sendResponse(List<MultipartFile> response, String id) {
-
-        centralClient.sendCommonEntityResponse(id, response.toArray(new MultipartFile[response.size()]));
-    }
+@Component
+public class CohortCharacterizationRequestHandler implements AtlasRequestHandler<CommonEntityDTO, List<MultipartFile>> {
+
+	private static final int PAGE_SIZE = 10000;
+	private static final String PACKAGE_NAME = "CohortCharacterization%d";
+	private final AtlasService atlasService;
+	private final GenericConversionService conversionService;
+	private final CommonEntityService commonEntityService;
+	private final Template runnerTemplate;
+	private final CentralSystemClient centralClient;
+
+	public CohortCharacterizationRequestHandler(AtlasService atlasService,
+																							GenericConversionService conversionService,
+																							CommonEntityService commonEntityService,
+																							@Qualifier("cohortCharacterizationTemplate") Template runnerTemplate,
+																							CentralSystemClient centralClient) {
+
+		this.atlasService = atlasService;
+		this.conversionService = conversionService;
+		this.commonEntityService = commonEntityService;
+		this.runnerTemplate = runnerTemplate;
+		this.centralClient = centralClient;
+	}
+
+	@Override
+	public List<CommonEntityDTO> getObjectsList(List<Atlas> atlasList) {
+
+		List<CohortCharacterization> ccList = atlasService.execute(atlasList, c -> c.getCohortCharacterizations(PAGE_SIZE).getContent());
```

Comment:
Can we add a check for Atlas version? If Atlas doesn't support CC then there is no reason to do query. Either immediately throw an exception (and handle it properly at UI) or, which I would do for now, just return empty list